controllers/ctltools.js

```diff
@@ -217,6 +218,12 @@ const showTools = (req, res) => {
         }
     }
 
+    let toolContextTypes;
+
+    api(req, {version: 'v3'}).get('/tools/context-types').then((contextTypes) => {
+        toolContextTypes = contextTypes.data;
+    })
+
```

Comment:
you should await this and handle errors. maybe use a Promise.all together with the external tools call and redirect to the error page in case of an error.
I think the `next()` function should lead to the error page.